model/src/main/kotlin/licenses/DefaultLicenseInfoProvider.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.licenses
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.utils.ProcessedDeclaredLicense
+
+/**
+ * The default [LicenseInfoProvider] that collects license information from an [ortResult].
+ */
+class DefaultLicenseInfoProvider(val ortResult: OrtResult) : LicenseInfoProvider {
+    private val licenseInfo: MutableMap<Identifier, LicenseInfo> = mutableMapOf()
```

Comment:
Thanks!

Reply:
If the log wasn't held during `createLicenseInfo` a deadlock would be impossible (to introduce)?
Consider changing that bit. 

It could be that just changing the type to `ConcurrentMap` would already solve the problem, but I haven checked its `getOrPut` implementation.

Reply:
I have change the type to `ConcurrentMap` and removed `@Synchronized`.